main.tf

```diff
@@ -305,8 +305,10 @@ module "graphdb" {
   ebs_default_kms_key = var.default_ebs_cmk_alias
 
   # DNS
-
-  route53_zone_dns_name = var.route53_zone_dns_name
+  route53_zone_enabled       = var.route53_zone_enabled
+  zone_dns_name              = var.route53_zone_enabled ? var.route53_zone_dns_name : var.existing_route53_zone_name
+  existing_route53_zone_id   = var.route53_zone_enabled ? null : var.route53_existing_dns_zone_id
```

Comment:
Aren't `route53_existing_zone_id` and `route53_existing_zone_name` better names?